howl

```diff
@@ -3,6 +3,13 @@
 set -euo pipefail
 trap 'echo failed: line $LINENO: $BASH_COMMAND' ERR
 
+# GitHub actions inputs starting with `INPUT_` take precedence over un-prefixed environment variables.
+SLACK_WEBHOOK_URL="${INPUT_SLACK_WEBHOOK_URL:-${SLACK_WEBHOOK_URL-}}"
```

Comment:
This is where you could use printenv, but I now realise you still need the `:-` part of the expansion. A helper function may be useful here instead:

```
input() {
    printenv "INPUT_${1//_/-}" || printenv "$1" || echo
}
```

and use it like:

    SLACK_WEBHOOK_URL=$(input SLACK_WEBHOOK_URL)


Now I've written this, I'm not sure I like it all that much but struggling to come up with anything better right now. Kebab case is idiomatic GHA YAML so it still feels like you should use it. :shrug:

Reply:
I like it. reads clearer in the script too:
```
SLACK_WEBHOOK_URL=$(input SLACK_WEBHOOK_URL)
SLACK_TEXT=$(input SLACK_TEXT)
SLACK_CHANNEL=$(input SLACK_CHANNEL)
DISCSORD_WEBHOOK_URL=$(input DISCSORD_WEBHOOK_URL)
DISCSORD_TEXT=$(input DISCSORD_TEXT)
```